File: backend/Dollars2.Api/Models/Transaction.cs
namespace Dollars2.Api.Models;

public class Transaction
{
    public int Id { get; set; }
    public int? AccountId { get; set; }
    public int UserId { get; set; }
    public string? ProviderTransactionId { get; set; }
    public DateTime Date { get; set; }
    public string Description { get; set; } = "";
    public decimal Amount { get; set; }
    public string? Notes { get; set; }
    public bool IsDeleted { get; set; }
    public bool IsPending { get; set; }
    public bool IsManual { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

File: frontend/src/types/transaction.ts
export interface TransactionResponse {
  id: number
  accountId: number | null
  accountName: string | null
  date: string
  description: string
  payee: string
  memo: string
  amount: number
  notes: string | null
  isDeleted: boolean
  isPending: boolean
  isManual: boolean
  assignments: TransactionAssignmentResponse[]
}

export interface TransactionAssignmentResponse {
  id: number
  lineItemId: number
  lineItemName: string
  amount: number
}

/** One archived sighting of a transaction, as the provider sent it during a sync run. */
export interface RawHistoryEntryResponse {
  syncedAt: string
  sourceType: string
  syncRunId: string
  /** The provider payload verbatim — a JSON string the server never re-parses. */
  rawJson: string
}
